internal/logging/defaults.go

```diff
@@ -20,18 +21,35 @@ import (
 // Logger describes a logging function, like Debug, Error, Warning, etc.
 type Logger func(msg string, args ...interface{})
 
+type safeBool struct {
+	mu sync.Mutex
```

Comment:
I would argue against using a mutex in the logger, as that is quite expensive to call.  My suggestions would be to 
- either use a regular boolean variable, as it would not be the end of the world if the verbose flag was incorrect in (and that is also a very unlikely situation) OR
- use an atomic bool like here: https://gist.github.com/mahan/6256149  That is much more efficient than a mutex...  It just ensures that we are not reading the value from a cached CPU register with an old value

Reply:
I opted not to use an atomic because it's more difficult to read the implementation and the performance improvement is measured in billionths of a second. Personally, I'd prefer the atomic op anyway, but felt strongly that Nathan wouldn't. Should I still proceed with the atomic?

Reply:
Regarding not using a guard: I strongly think we should not accept races as "benign". The concern is not simply incorrect data, it's torn data which is "undefined" behavior and can pop the application.

Reply:
Let's just leave the mutex. I always strive to reduce locks if it is easy, but it is not the end of the world - especially in our use case.